python/paddle/fluid/tests/unittests/test_auto_parallel_partitioner.py

```diff
@@ -135,6 +135,110 @@ def initialization_check(mode, dist_context, dist_startup_prog,
+                in_var)
+            tensor_dims_mapping = tensor_dist_attr.get_dims_mapping()
+            in_var_dims_mapping = op_dist_attr.get_input_dims_mapping(
+                in_varname)
+            if tensor_dims_mapping != in_var_dims_mapping:
+                equal = False
+        for out_varname in dist_ops[i].desc.output_arg_names():
+            out_var = dist_main_prog.global_block().var(out_varname)
+            tensor_dist_attr = dist_context.get_tensor_distributed_attr_for_program(
+                out_var)
+            tensor_dims_mapping = tensor_dist_attr.get_dims_mapping()
+            out_var_dims_mapping = op_dist_attr.get_output_dims_mapping(
+                out_varname)
+            if tensor_dims_mapping != out_var_dims_mapping:
+                equal = False
+
+        dist_op_process_mesh = op_dist_attr.get_process_mesh()
+        dist_op_impl_idx = op_dist_attr.get_impl_idx()
+        if serial_op.desc.id() == dist_ops[i].desc.id() or \
+            serial_process_mesh != dist_op_process_mesh or \
+            serial_impl_idx != dist_op_impl_idx:
+            equal = False
+
+    return equal
+
+
+def distributed_attr_check_for_dist_op(serial_main_prog, dist_main_prog,
+                                       dist_context, serial_op_idx,
+                                       dist_op_idx):
+
+    equal = True
+    serial_ops = serial_main_prog.global_block().ops
+    dist_ops = dist_main_prog.global_block().ops
+
+    for i in range(len(serial_op_idx)):
+        serial_op = serial_ops[serial_op_idx[i]]
+        dist_op_0 = dist_ops[dist_op_idx[i][0]]
+        if dist_op_0.type == "c_identity":
+            # serial op input's dist_attr
+            serial_in_dist_attr = get_input_var_dist_attr(
+                serial_op, serial_main_prog, dist_context)
+            # c_identity output's(new var) dist_attr
+            identity_out_dist_attr = get_output_var_dist_attr(
+                dist_op_0, dist_main_prog, dist_context)
+            # check var dist_attr
+            equal = check_equal_var_dist_attr(serial_in_dist_attr,
+                                              identity_out_dist_attr)
+        else:
+            # serial op output's dist_attr
+            serial_out_dist_attr = get_output_var_dist_attr(
+                serial_op, serial_main_prog, dist_context)
+            # dist op output's(new var) dist_attr
+            out_dist_attr = get_output_var_dist_attr(dist_op_0, dist_main_prog,
+                                                     dist_context)
+            # check var dist_attr
+            equal = check_equal_var_dist_attr(serial_out_dist_attr,
+                                              out_dist_attr)
+
+        # check op's dist_attr 
+        equal = check_equal_dist_op_attr(dist_context, dist_main_prog,
+                                         serial_op, dist_ops, dist_op_idx[i])
+
+    return equal
+
+
```

Comment:
you should add a new test for all case: checking all vars and  ops have dist attr

Reply:
done.